src/common/column/src/types/native.rs

```diff
@@ -524,9 +534,27 @@ impl timestamp_tz {
     }
 
     #[inline]
-    pub fn try_total_micros(&self) -> Option<i64> {
+    pub fn try_timestamp_with_offset(&self) -> Option<i64> {
         let offset_micros = self.micros_offset()?;
-        self.timestamp().checked_sub(offset_micros)
+        self.timestamp().checked_add(offset_micros)
+    }
+
+    #[inline]
+    fn encode_utc(timestamp: i64, offset: i32) -> i64 {
+        let Some(offset_micros) = Self::micros_offset_inner(offset as i64) else {
+            error!(
+                "timestamp offset is out of range: timestamp={}, offset={}",
+                timestamp, offset
+            );
+            return 0;
+        };
+        timestamp.checked_sub(offset_micros).unwrap_or_else(|| {
```

Comment:
Why use timestamp subtracte offset? These two numbers should be independent and stored separately.